aten/src/ATen/native/cuda/ForeachTensorAddList.cu

```diff
@@ -0,0 +1,47 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/ForeachUtils.h>
+#include <ATen/native/cuda/ForeachFunctors.cuh>
+
+namespace at { namespace native {
+
+std::vector<Tensor> foreach_tensor_add_list_kernel_cuda(TensorList tensors1, TensorList tensors2) {
+    verify_list(tensors1, tensors2);
+
+    if (!check_fast_route(tensors1, tensors2)) {
```

Comment:
nit: As a function name, "check_fast_route" doesn't sound like it would return a boolean (it sounds like it would return nothing). Something like "can_use_fast_route" sounds like it would return a boolean.